uiServices/src/testSuiteServices.go

```diff
@@ -30,58 +41,98 @@ func postTestSuites(rw http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	if !ValidateFileNameAndHeader(rw,req,configPathDir ,testSuite.Name){
+	if !ValidateFileNameAndHeader(rw,req, testSuitePathDir,testSuite.Name){
 		return
 	}
 
-	if !validateTestSuiteJsonWithSchema(buf.Bytes()) {
+	if !ValidateJsonWithSchema(buf.Bytes(), "testSuite_schema.json", "TestSuite") {
 		rw.WriteHeader(http.StatusBadRequest)
 		return
 	}
 
-	if !FilePathExist(configPathDir) {
-		logrus.Error("File path does not exist", err)
+	if !FilePathExist(testSuitePathDir) {
+		logrus.Error("Directory path does not exist", err)
 		rw.WriteHeader(http.StatusBadRequest)
 		return
 
 	}
 
-	if FilePathExist(configPathDir + testSuite.Name + ".xml") {
+	if FilePathExist(testSuitePathDir + testSuite.Name + ".xml") {
 		logrus.Error("File already exists")
 		rw.WriteHeader(http.StatusBadRequest)
 		return
 
 	}
 
-	if !testSuiteWriterXml(testSuite, configPathDir+testSuite.Name+".xml") {
+	if !testSuiteWriterXml(testSuite, testSuitePathDir +testSuite.Name+".xml") {
 		rw.WriteHeader(http.StatusInternalServerError)
 		return
 	}
 
 	rw.WriteHeader(http.StatusCreated)
 }
 
-func validateTestSuiteJsonWithSchema(testSuite []byte) bool {
+func ValidateJsonWithSchema(testSuite []byte, schemaName, structType string) bool {
 	goPath := os.Getenv("GOPATH")
-	schemaLoader := gojsonschema.NewReferenceLoader("file:///" + goPath + "/src/github.com/xtracdev/automated-perf-test/ui-src/src/assets/testSuite_schema.json")
+	schemaLoader := gojsonschema.NewReferenceLoader("file:///" + goPath + "/src/github.com/xtracdev/automated-perf-test/ui-src/src/assets/"+ schemaName)
 	documentLoader := gojsonschema.NewBytesLoader(testSuite)
 	logrus.Info(schemaLoader)
 	result, error := gojsonschema.Validate(schemaLoader, documentLoader)
 
 	if error != nil {
 		return false
 	}
-	if result.Valid() {
-		logrus.Info("**** The TestSuite document is valid *****")
-
-		return true
-	}
 	if !result.Valid() {
-		logrus.Error("**** The TestSuite document is not valid. see errors :")
+		logrus.Error("**** The "+structType+" document is not valid. see errors :")
 		for _, desc := range result.Errors() {
 			logrus.Error("- ", desc)
 			return false
 		}
 	}
+
+	logrus.Info("**** The "+structType+" document is valid *****")
+
 	return true
 }
+
+
+func putTestSuites(rw http.ResponseWriter, req *http.Request) {
+	path := getTestSuiteHeader(req)
+	testSuiteName := chi.URLParam(req, "testSuiteName")
+
+	if !ValidateFileNameAndHeader(rw, req, path, testSuiteName) {
+		return
+	}
+
+	testSuitePathDir := fmt.Sprintf("%s%s.xml", path, testSuiteName)
+	buf := new(bytes.Buffer)
+	buf.ReadFrom(req.Body)
+
+	if !ValidateJsonWithSchema(buf.Bytes(),"testSuite_schema.json", "TestSuite") {
+		rw.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	testSuite := testStrategies.TestSuite{}
+	err := json.Unmarshal(buf.Bytes(), &testSuite)
+	if err != nil {
+		logrus.Error("Cannot Unmarshall Json")
+		rw.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	if !FilePathExist(testSuitePathDir) {
```

Comment:
I kind of think these  if stmts should be move up closer to line 107, the idea being that we fail quickly instead of carrying out operations only to find that they were not needed.